python/couler/couler/argo.py

```diff
@@ -527,7 +527,8 @@ def _dump_yaml():
     if len(_secrets) > 0:
```

Comment:
If `dry_run` is `False`, the `_secrets` is `None`. Then, does `len(_secrets)` leaders to an Exception?
https://github.com/sql-machine-learning/sqlflow/blob/3e11d03c0694b75a6cd8e34ac52847931bf3a647/python/couler/couler/argo.py#L741-L743

Reply:
May not, `_secrets` would be initialized as  an empty dict:

https://github.com/sql-machine-learning/sqlflow/blob/3e11d03c0694b75a6cd8e34ac52847931bf3a647/python/couler/couler/argo.py#L23